### ISSUE TITLE
Download fails with curl usage error on ZSH

### ISSUE THREAD

**Issue by marshallpierce** [sdkman/sdkman-cli#531]
With 5.1.18+191 and curl 7.52.1, I get:

```
% sdk install gradle

Downloading: gradle 3.3

In progress...

curl: option --progress-bar --location: is unknown
curl: try 'curl --help' or 'curl --manual' for more information
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
```

However, when I run `curl --help`, I do see `--progress-bar` and `--location` as supported options, and they work when I try them. `which -a curl` shows only `/usr/bin/curl`.

**Comment by ajmath** [comment]
Also seeing this with 5.5.1+179 and curl 7.51 on macOS Sierra with zsh.  Switching to bash I did not experience this.

**Comment by brandon-fryslie** [comment]
I can verify that using Bash instead of Zsh fixed this issue for me.

**Comment by ahaasler** [comment]
Also seeing this in 5.5.2+183 and curl 7.52.1 on Arch with zsh.

**Comment by marc0der** [comment]
Seems very odd: I also use the same version 5.5.2+183 and with curl 7.52.1 on Arch with zsh without any issues.

I suspect that this might be the result of an interaction with something else being used/loaded in the zsh shell. I personally use oh-my-zsh. Is anyone here perhaps using prezto? Prezto is know to cause issues with sdkman and many others since it makes very rude assumptions about shell defaults.

**Comment by marshallpierce** [comment]
I don't use either oh-my-zsh or prezto or any other pre-canned zsh config system.

This has to do with the way word splitting is done in variable interpolation. In [sdkman-utils.sh](https://github.com/sdkman/sdkman-cli/blob/master/src/main/bash/sdkman-utils.sh#L34), it does this:

```
    local curl_params="--progress-bar --location"
    ...
    curl ${curl_params} "$1"
```

And expects that to be treated as *two* params. It's not, so curl complains.

This is probably related to https://github.com/sdkman/sdkman-cli/issues/210, which I submitted a PR for _four years ago_...

**Comment by marshallpierce** [comment]
After the fix for `shwordsplit` in https://github.com/sdkman/sdkman-cli/commit/ab14a419f9cf13dd07174580023f7106f656c9c5 this is still broken. It will probably be another zsh/bash conditional. This syntax will make zsh do the "string split on space when interpolating" that you want:

```
curl ${=curl_params} "$1"
```

**Comment by marc0der** [comment]
I managed to replicate this with vanilla ZSH running in Docker. I've deployed the fix to our beta channel for testing. I seems to work fine for me now. @marshallpierce can you confirm that it's working for you now?

**Comment by ovovchuk** [comment]
@marc0der I have tried your fix and it worked fine for me with zsh shell.

**Comment by mikezx6r** [comment]
I can also confirm this works. Mac OS X 10.11.6, zsh, zprezto.

Was experiencing issue before updating to the beta channel. Install worked after upgrade.

**Comment by marshallpierce** [comment]
Yep, works for me too!

**Comment by marc0der** [comment]
Great, has been released as 5.5.7.
